scripts/clan_resources/herb/herb_supply.py

```diff
@@ -0,0 +1,728 @@
+
+        return list_of_herb_strs, found_herbs
+
+    def _add_collection_to_storage(self, med_cats):
+        """
+        Adds herbs in self.collected to self.storage and clears self.collected dict. If Clan has a CAMP skill med,
+        this is where an expiration perk is added.
+        """
+        # add empty "clump" to all uncollected herb stores
+        for herb in [x for x in self.base_herb_list if x not in self.collected and x in self.storage]:
+            self.storage.get(herb, []).insert(0, 0)
+
+        for med in med_cats:
+            # check if any meds can use a skill to store herbs better (aka, reduce time to expire)
+            # we base this modifier on their path points,
+            # this means there's skill variation even within cats with matching tiers
+            if med.skills.primary.path == SkillPath.CAMP:
+                modifier = med.skills.primary.points
+            elif med.skills.secondary and med.skills.secondary.path == SkillPath.CAMP:
+                modifier = med.skills.secondary.points
+            else:
+                continue
+
+            better_storage = []
+            for herb, count in self.collected.items():
+                if not self.storage.get(herb, []):
+                    # herbs can't be stored better if there isn't an existing store of that herb
+                    break
+
+                # attempt the better storage
+                if randint(1, 35 - modifier) == 1:
+                    better_storage.append(herb)
+                    self.storage[herb][0] += count
+                    continue
+
+            if better_storage:
+                # inform player of expiration perk
+                self.log.append(i18n.t(
+                    "screens.med_den.log_messages.better_storage",
+                    name=med.name,
+                    herbs=adjust_list_text([self.herb[herb].plural_display for herb in better_storage])
+                ))
+                # remove herbs that were stored well from the collection
+                for herb in better_storage:
+                    self.collected.pop(herb)
+
+        # store if no meds managed to store better
+        for herb, count in self.collected.items():
+            if herb in self.storage:
+                self.storage.get(herb, []).insert(0, count)
+            else:
+                self.storage[herb] = [count]
+
+        # clear collection dict
+        self.collected = {}
+
+    def _use_herbs(self, treatment_cat):
+        """
+        utilize current herb supply on given condition
+        """
+        # collate all cat's conditions
+        condition_dict = treatment_cat.injuries.copy()
+        condition_dict.update(treatment_cat.illnesses)
+        condition_dict.update(treatment_cat.permanent_condition)
+
+        # collate all the source info for conditions
+        source_dict = ILLNESSES.copy()
+        source_dict.update(INJURIES)
+        source_dict.update(PERMANENT)
```

Comment:
Can we store this somewhere rather than creating it every time (perhaps in `utility.py`?